pandas/tseries/index.py

```diff
@@ -439,7 +439,7 @@ def _generate(cls, start, end, periods, name, offset,
                 tz = tz.localize(date.replace(tzinfo=None)).tzinfo
 
         if tz is not None and inferred_tz is not None:
-            if not inferred_tz == tz:
+            if not tslib.get_timezone(inferred_tz) == tslib.get_timezone(tz):
```

Comment:
hmm, can you repro that directly via a DatetimeIndex construction call?

does it break anything else?